transmart-core-db/grails-app/services/org/transmartproject/db/support/ChoppedInQueryCondition.groovy

```diff
@@ -0,0 +1,108 @@
+package org.transmartproject.db.support
+
+import org.apache.commons.lang.StringEscapeUtils
+import grails.orm.HibernateCriteriaBuilder
+import org.grails.datastore.mapping.query.api.Criteria
+import org.hibernate.criterion.Restrictions
+import org.transmartproject.core.exceptions.InvalidRequestException
+
+/**
+ * This class aimed to overcome oracle limitation on the number of items in IN clause:
+ * Message: ORA-01795: maximum number of expressions in a list is 1000
+ *
+ * This class constructs composite condition
+ * Instead of making one condition
+ * patient_num IN (1, 2, 3, ... 1200)
+ * it chops list of items and join sub-conditions with OR operations
+ * patient_num IN (1, 2, 3, ... 1000) OR patient_num IN (1001, 1002, 1003, ... 1200)
+ */
+class ChoppedInQueryCondition {
+
+    static final int MAX_LIST_SIZE = 1000
+    static final String ALWAYS_TRUE_CONDITION = '1=1'
+    static final String ALWAYS_FALSE_CONDITION = '0=1'
+
+    private final String fieldName
+    private final List inItems
+
+    ChoppedInQueryCondition(String fieldName, List inItems) {
+        this.fieldName = fieldName
+        this.inItems = inItems
+    }
+
+    @Lazy
+    Map parametersValues = {
+        if (!inItems) return [:]
+        def chunks = inItems.collate(MAX_LIST_SIZE)
+        (0..<chunks.size()).collectEntries { index -> ["_${index}".toString(), chunks[index]] }
+    }()
+
+    @Lazy
+    String queryConditionTemplate = {
```

Comment:
Are these actually used and, if so, can it be prevented? Having a list of chunks instead of map is preferred.